### ISSUE TITLE
change currencies source to iban

### ISSUE THREAD

**Issue by ecocode** [finance-quote/finance-quote#126]
currencies are still retrieved from yahoo, which fails.
we can use https://www.iban.com/currency-codes as a new source for this.

**Comment by vincentl** [comment]
Since several of the other modules I recently worked on required table parsing, I made a first attempt at addressing this issue.

t/currencies.t passes
t/currency_lookup.t passes
t/currency.t fails

Need to investigate the failed test and if there are other currency dependencies before closing issue.

**Comment by bpschuck** [comment]
I believe I figured this out.

If you look at the %currencies hash, even though there is a field named 'number', it's populated with the code.

    'AED' => {
        'country' => ['UNITED ARAB EMIRATES (THE)'],
        'name'    => 'UAE Dirham',
        'number'  => 'AED'
    },

However, in the fetch_live_currencies function, the 'number' field in each hash is being assigned what is in the 'number' column returned from the source URL.

    $result{$code} = {'name'    => $currency,
                        'country' => [$country],
                        'number'  => $number};

While testing this I also discovered that sometimes the country names returned from the source URL have extraneous spaces (usually 3 extra spaces). Example:

<PRE>
"HEARD ISLAND AND McDONALD    ISLANDS" instead of 
"HEARD ISLAND AND McDONALD ISLANDS".
</PRE>

I just changed `'number'  => $number` to `'number'  => $code` and added the extra spaces in the %currencies hash. All three test files return okay on all tests.
